Library/Homebrew/cmd/outdated.rb

```diff
@@ -28,10 +28,19 @@ def outdated
 
   def print_outdated(formulae)
     verbose = ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+    fetch_head = ARGV.fetch_head?
 
-    formulae.select(&:outdated?).each do |f|
+    outdated_formulae = formulae.select { |f| f.outdated?(:fetch_head => fetch_head) }
+
+    outdated_formulae.each do |f|
       if verbose
-        puts "#{f.full_name} (#{f.outdated_versions*", "} < #{f.pkg_version})"
+        outdated_versions = f.outdated_versions(:fetch_head => fetch_head)
+        if outdated_versions.any? && f.head? && !fetch_head \
+            || (version = f.latest_head_version) && f.tab_version_outdated?(version)
+          puts "#{f.full_name} (#{outdated_versions.join(", ")}) < latest HEAD"
+        else
+          puts "#{f.full_name} (#{outdated_versions.join(", ")}) < #{f.pkg_version}"
+        end
```

Comment:
This part is overcomplicated, you should use the same logic as below `print_outdated_json`, i.e.
```ruby
      current_version = if f.head? && outdated_versions.any? { |v| v.to_s ==  f.pkg_version.to_s }
        "latest HEAD"
      else
        f.pkg_version.to_s
      end
      puts "#{f.full_name} (#{outdated_versions.join(", ")}) < #{current_version}"

```